skiko/src/jvmMain/cpp/windows/directXRedrawer.cc

```diff
@@ -0,0 +1,197 @@
+#ifdef SK_DIRECT3D
+#include <windows.h>
+#include <jawt_md.h>
+#include "GrBackendSurface.h"
+#include "GrDirectContext.h"
+#include "SkSurface.h"
+
+#include "d3d/GrD3DTypes.h"
+#include <d3d12sdklayers.h>
+#include "d3d/GrD3DBackendContext.h"
+#include <d3d12.h>
+#include <dxgi1_4.h>
+
+#define GR_D3D_CALL_ERRCHECK(X)                                        \
+    do                                                                 \
+    {                                                                  \
+        HRESULT result = X;                                            \
+        SkASSERT(SUCCEEDED(result));                                   \
+        if (!SUCCEEDED(result))                                        \
+        {                                                              \
+            SkDebugf("Failed Direct3D call. Error: 0x%08x\n", result); \
+        }                                                              \
+    } while (false)
+
+extern "C" jboolean Skiko_GetAWT(JNIEnv *env, JAWT *awt);
```

Comment:
We could move it to some common header, i.e. `jni_helpers.h`.

Reply:
This is my mistake - we don't need `Skiko_GetAWT` in this file. Removed.